app/schemas/user_signup.js

```diff
@@ -0,0 +1,38 @@
+const db = require('../models');
+
+const signUpSchema = {
+  firstName: {
+    exists: true,
+    notEmpty: true,
+    errorMessage: 'First name cannot be empty!'
+  },
+  lastName: {
+    exists: true,
+    notEmpty: true,
+    errorMessage: 'Last name cannot be empty!'
+  },
+  email: {
+    exists: true,
+    isEmail: { errorMessage: 'Not a valid email.' },
+    contains: { options: '@wolox.co', errorMessage: 'Email does not belong wolox domain.' },
+    notEmpty: { errorMessage: 'email cannot be empty!' },
+    custom: {
+      options: async value => {
+        const user = await db.User.findOne({ where: { email: value } });
+        if (user) {
+          throw new Error('Email is already registered.');
```

Comment:
This validator is nice, but I think throwing this error like this would make it have an 500 http code, which is not very telling. Better to throw a custom error here :)

Reply:
this wil throw a validation error with the code stablish for that, previosly 409 but i'm going to change it. If a different code must be return for dis particular case tell and i will make it. :)

Reply:
The thing is, `409` is the right code when the email already exists but not for when an input is incorrect, that would be `422`